src/stepFunctions/wizards/previewStateMachineCDKWizard.ts

```diff
@@ -3,238 +3,87 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import * as vscode from 'vscode'
 import * as nls from 'vscode-nls'
-import * as picker from '../../shared/ui/picker'
-
 const localize = nls.loadMessageBundle()
 
-import { AppNode } from '../../cdk/explorer/nodes/appNode'
-import { CdkAppLocation } from '../../cdk/explorer/cdkProject'
-import { ConstructNode } from '../../cdk/explorer/nodes/constructNode'
+import * as vscode from 'vscode'
+import * as path from 'path'
+import { CdkAppLocation, getApp } from '../../cdk/explorer/cdkProject'
+import { ConstructNode, isStateMachine } from '../../cdk/explorer/nodes/constructNode'
 import { detectCdkProjects } from '../../cdk/explorer/detectCdkProjects'
-import {
-    MultiStepWizard,
-    WIZARD_GOBACK,
-    WIZARD_TERMINATE,
-    wizardContinue,
-    WizardStep,
-} from '../../shared/wizards/multiStepWizard'
-
-export interface CdkAppLocationPickItem {
-    label: string
-    cdkApplocation: CdkAppLocation | undefined
-}
-
-export interface TopLevelNodePickItem {
-    label: string
-    topLevelNode: ConstructNode | undefined
-}
-
-export interface ConstructNodePickItem {
-    label: string
-    stateMachineNode: ConstructNode | undefined
-}
-
-interface PreviewStateMachineCDKWizardResponse {
-    cdkApplication: CdkAppLocationPickItem
-    topLevelNode: TopLevelNodePickItem
-    stateMachine: ConstructNodePickItem
+import { Wizard, WIZARD_BACK } from '../../shared/wizards/wizard'
+import { createQuickPick } from '../../shared/ui/pickerPrompter'
+import { createCommonButtons } from '../../shared/ui/buttons'
+import { ConstructTreeEntity } from '../../cdk/explorer/tree/types'
+import { getDisplayLabel } from '../../cdk/explorer/tree/treeInspector'
+
+function createLocationPrompter() {
```

Comment:
I chose to rewrite the wizard. Would've taken too much time to surgically correct it plus the tests. I ended up removing one step: users shouldn't need to select individual stacks, just show all state machines.